### PR TITLE
Publish to IIS websites other than "Default Web Site".

### LINKED ISSUE #944: Allow publication of multiple applications to the same GCE VM.
Feature request stemming from [this](https://stackoverflow.com/questions/50071496/visual-studio-net-mvc-deploy-to-google-cloud-into-specific-site-on-iis) SO question.

We currently only support publication to the default IIS application on a GCE VM. The requests is for us to allow publication of different IIS applications on the same VM.

### PR DESCRIPTION
**Pull request by ILMTitan**
Change StatusbarHelper into an MEF service.
Add Site Name input to Publish to Google Compute Engine VM dialog.
Change the WindowsVmDeployment, making it an MEF service, making it simpler and having it take a Site Name.
Fixes #944

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionUnitTests.csproj [hunk 1]
```diff
@@ -385,9 +389,7 @@
   <ItemGroup>
     <Analyzer Include="..\packages\Microsoft.VisualStudio.Threading.Analyzers.15.7.18\analyzers\cs\Microsoft.VisualStudio.Threading.Analyzers.dll" />
   </ItemGroup>
-  <ItemGroup>
-    <Folder Include="Deployment\" />
-  </ItemGroup>
+  <ItemGroup />
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionUnitTests.csproj]: Remove

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/Deployment/MSBuildPropertyTests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        [TestMethod]
+        public void TestToString_ReturnsMSBuildCliArg()
+        {
+            var objectUnderTest = new MSBuildProperty(ExpectedPropertyName, ExpectedPropertyValue);
+
+            Assert.AreEqual("/p:ExpectedPropertyName=\"Expected Property Value\"", objectUnderTest.ToString());
+        }
+    }
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/Deployment/MSBuildPropertyTests.cs]: Worth adding tests for property/value being null

**Comment by amanda-tarafa** [comment]
Let me know when you have released this so I can update the answer on SO. Cheers!
